packages/toolkit/src/createSlice.ts

```diff
@@ -585,17 +849,167 @@ function getType(slice: string, actionKey: string): string {
   return `${slice}/${actionKey}`
 }
 
-interface BuildCreateSliceConfig {
-  creators?: {
-    asyncThunk?: typeof asyncThunkCreator
+export const reducerCreator: ReducerCreator<ReducerType.reducer> = {
+  type: ReducerType.reducer,
+  create(caseReducer: CaseReducer<any, any>) {
+    return Object.assign(
+      {
+        // hack so the wrapping function has the same name as the original
+        // we need to create a wrapper so the `reducerDefinitionType` is not assigned to the original
+        [caseReducer.name](...args: Parameters<typeof caseReducer>) {
+          return caseReducer(...args)
+        },
+      }[caseReducer.name],
+      {
+        _reducerDefinitionType: ReducerType.reducer,
+      } as const,
+    )
+  },
+  handle({ type, reducerName }, reducer, context) {
+    context
+      .addCase(type, reducer as any)
```

Comment:
:hand: this seems like a common sequence to do. Worth building it in?

Reply:
hmm, what would it be called?
currently a custom creator can just import reducerCreator and use it directly:

```ts
reducerCreator.handle(
  { type, reducerName },
  reducerCreator.create(reducer),
  context,
);
```

I do this in my history-adapter creator: https://github.com/EskiMojo14/history-adapter/blob/main/src/creator.ts#L122-L126